src/frontend/JasmineGraphFrontEnd.cpp

```diff
@@ -1314,1497 +1420,126 @@ bool JasmineGraphFrontEnd::isGraphActive(std::string graphID, SQLiteDBInterface
-            int workerPort = atoi(string(currentWorker.port).c_str());
-            int workerDataPort = atoi(string(currentWorker.dataPort).c_str());
-            frontend_logger.log("====>PORT:" + std::to_string(workerPort), "info");
-
-            partitionId = *partitionIterator;
-            intermRes.push_back(
-                    std::async(std::launch::async, JasmineGraphFrontEnd::getTriangleCount, atoi(graphId.c_str()), host,
-                               workerPort, workerDataPort, atoi(partitionId.c_str()), masterIP, uniqueId, isCompositeAggregation));
-        }
-    }
-
-    for (auto &&futureCall:intermRes) {
-        result += futureCall.get();
-    }
-
-    if (!isCompositeAggregation) {
-        long aggregatedTriangleCount = JasmineGraphFrontEnd::aggregateCentralStoreTriangles(sqlite, graphId,masterIP);
-        result += aggregatedTriangleCount;
-        frontend_logger.log("###FRONTEND### Getting Triangle Count : Completed: Triangles " + to_string(result),
-                            "info");
-    }
-
-    auto end = chrono::high_resolution_clock::now();
-    auto dur = end - begin;
-    auto msDuration = std::chrono::duration_cast<std::chrono::milliseconds>(dur).count();
-
-    std::string durationString = std::to_string(msDuration);
-
-    if (canCalibrate) {
-        Utils::updateSLAInformation(perfSqlite, graphId, partitionCount, msDuration, TRIANGLES, Conts::SLA_CATEGORY::LATENCY);
-    }
-
 
-    triangleTree.clear();
-    combinationWorkerMap.clear();
-
-    return result;
+void JasmineGraphFrontEnd::getAndUpdateUploadTime(std::string graphID, SQLiteDBInterface sqlite) {
+    struct tm tm;
+    vector<vector<pair<string, string>>> uploadStartFinishTimes = sqlite.runSelect(
+            "SELECT upload_start_time,upload_end_time FROM graph WHERE idgraph = '" + graphID + "'");
+    string startTime = uploadStartFinishTimes[0][0].second;
+    string endTime = uploadStartFinishTimes[0][1].second;
+    string sTime = startTime.substr(startTime.size() - 14, startTime.size() - 5);
+    string eTime = endTime.substr(startTime.size() - 14, startTime.size() - 5);
+    strptime(sTime.c_str(), "%H:%M:%S", &tm);
+    time_t start = mktime(&tm);
+    strptime(eTime.c_str(), "%H:%M:%S", &tm);
+    time_t end = mktime(&tm);
+    double difTime = difftime(end, start);
+    sqlite.runUpdate("UPDATE graph SET upload_time = " + to_string(difTime) + " WHERE idgraph = " + graphID);
+    frontend_logger.log("Upload time updated in the database", "info");
 }
 
+JasmineGraphHashMapCentralStore JasmineGraphFrontEnd::loadCentralStore(std::string centralStoreFileName) {
+    frontend_logger.log("Loading Central Store File : Started " + centralStoreFileName,"info");
+    JasmineGraphHashMapCentralStore *jasmineGraphHashMapCentralStore = new JasmineGraphHashMapCentralStore();
+    jasmineGraphHashMapCentralStore->loadGraph(centralStoreFileName);
+    frontend_logger.log("Loading Central Store File : Completed","info");
```

Comment:
Can we also add the name of the central store file here in the log message? Otherwise we do not know which central store got loaded.

Reply:
It is already logged in line 1442

Reply:
Ok